ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -146,63 +146,68 @@ Tensor convert_float_vector_to_tt_tensor(
 }
 
 Tensor create_tt_tensor_from_py_data(
-    std::size_t num_elements,
     std::size_t py_data_ptr,
-    const ttnn::SmallVector<uint32_t>& shape,
-    const DataType data_type,
-    const std::optional<Tile>& optional_tile,
+    const TensorSpec& tensor_spec,
+    Device* device,
     bool enable_borrow,
     const std::function<void()>& on_creation_callback = [] {},
     const std::function<void()>& on_destruction_callback = [] {}) {
+    auto layout = tensor_spec.layout();
+    if (layout != Layout::ROW_MAJOR) {
+        enable_borrow = false;
```

Comment:
enable_borrow is a param here. if you set it to false - you override user request. I think this must be an exception or in the worst case a warning

Reply:
This flag is poorly implemented. It's actually always default true (ie. user never specifies for it to be true) and for multi-device tensors, it asks for false.

Reply:
i think it's better to make it a bool with default value and use it to calculate another internal bool